docs/designs/045-node-validation.md

```
@@ -0,0 +1,445 @@
+* **Validating existing nodes:** as discussed above, existing nodes will be validated post remediation by having the fault-quarantine module request an existing node is validated.
+* **Supporting multiple clients:** we would be able to support multiple clients requesting validation by having each client interact with the HealthEvent API.
+* **Overlapping validations:** we will allow multiple clients to write events requesting validation for the same node. The validation-controller will be responsible for de-duplicating or combining tests from multiple events targeting the same node. For example, if a validation was requested by both fault-quarantine and for a new node, the controller would be responsible for handling the duplicated requests.
+* **Validating multiple nodes:** this approach does not allow a client to specify that a group of nodes should be validated and will require 1 HealthEvent per node per GPU needing validation.
+
+**Conclusion:** this approach would be recommended if all validation clients are in NVSentinel, we accept a HealthEvent API and MongoDB dependency on validation, and if we only need to support declaring individual nodes as needing validation.
+
+**Option 3: CRD as the validation trigger (recommended)**
+
+Rather than require clients to update node object status conditions and annotations or rely on the HealthEvent API, we could provide a ValidationRequest CRD to allow a client to declare that a particular node or a group of nodes should be validated.
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)